### DISCUSSION TITLE
How to Constrain Cost Dimension with FIXED and Variable Vehicle Cost?

### DISCUSSION THREAD

**Discussion by FreddyHunter** [google/or-tools#2729] [Routing (and legacy CP) questions]
Hi Guys,

I am using fixed and variable costs in my model, and I also created a Dimension Cost to limit the maximum cost allowed per vehicle itinerary. For that, I am using AddDimensionWithVehicleTransits once my route per vehicle can vary, but it is taking in consideration just the variable cost, that is, it is not sum the fixed cost on dimension. Does any have any suggestions how to model it?

I am coding in Python and using latest version of ortools.

Thanks in advance.

**Comment by Mizux** [comment]
What do you mean by "it is not sum the fixed cost on dimension".

Did you mean the dimension end's node CumulVar values are not part of the objective ? If yes then you should try to use
https://github.com/google/or-tools/blob/b37d9c786b69128f3505f15beca09e89bf078a89/ortools/constraint_solver/routing.h#L2482-L2489

**Reply by FreddyHunter** [reply]
Hi Mizux, 
I set fixed costs for each vehicle and variable costs per route performed by each vehicle. What I was expecting to have was: if the optimizer decided to use vehicle X it would need to consider its fixed cost + variable cost * distance at the cost dimension constraints. For example, if I want the route performed by vehicle X costs a maximum of 1000 the cost dimension would consider 100 (for example as fixed cost) and 900 as variable cost (considering 1 unit per km performed. Below is the way I modeled the cost dimension:

routing.AddDimensionWithVehicleTransits(
    transit_callback_index,
    0,  
    data['vehicle_max_costs'], 
    True, 
    'Cost')

The transit_callback_index is being calculated getting the distance between from_node to_node multiplied by variable cost using SetArcCostEvaluatorOfVehicle. The fixed cost I am setting using SetFixedCostOfVehicle. But, when I run the optimizer the vehicle cost per route is getting higher than 1000 for example. I mean, the optimizer is leaving the vehicle perform 1000 km and not 900 (considering that 100 is fixed cost). Is it clear? And I am not using none SetSpanCostCoefficient.
Thanks.
